tasks-by-users/anask601/02-js-monster-killer-game/assets/scripts/app.js

```diff
@@ -0,0 +1,191 @@
+      finalPlayerHealth: playerHealth,
+    };
+  } else if (ev === LOG_EVENT_GAME_OVER) {
+    logEntry = {
+      event: ev,
+      value: val,
+      finalMonsterHealth: monsterHealth,
+      finalPlayerHealth: playerHealth,
+    };
+  }
+  battleLog.push(logEntry);
+}
+
+function reset() {
+  currentMonsterHealth = chosenMaxLife;
+  currentPlayerHealth = chosenMaxLife;
+  resetGame(chosenMaxLife);
+}
+
+function endRound() {
+  const initialPlayerHealth = currentPlayerHealth;
+  const playerDamage = dealPlayerDamage(MONSTER_ATTACK_VALUE);
+  currentPlayerHealth -= playerDamage;
+  writeToLog(
+    LOG_EVENT_MONSTER_ATTACK,
+    playerDamage,
+    currentMonsterHealth,
+    currentPlayerHealth
+  );
+
+  if (currentPlayerHealth <= 0 && hasBonusLife) {
+    hasBonusLife = false;
+    removeBonusLife();
+    currentPlayerHealth = initialPlayerHealth;
+    setPlayerHealth(initialPlayerHealth);
+    alert("You would be dead but the bonus life saved you!");
+  }
+
+  if (currentMonsterHealth <= 0 && currentPlayerHealth > 0) {
+    alert("You won!");
+    writeToLog(
+      LOG_EVENT_GAME_OVER,
+      "PLAYER WON",
+      currentMonsterHealth,
+      currentPlayerHealth
+    );
+  } else if (currentPlayerHealth <= 0 && currentMonsterHealth > 0) {
+    alert("You lost!");
+    writeToLog(
+      LOG_EVENT_GAME_OVER,
+      "MONSTER WON",
+      currentMonsterHealth,
+      currentPlayerHealth
+    );
+  } else if (currentPlayerHealth <= 0 && currentMonsterHealth <= 0) {
+    alert("You have a draw!");
+    writeToLog(
+      LOG_EVENT_GAME_OVER,
+      "A DRAW",
+      currentMonsterHealth,
+      currentPlayerHealth
+    );
+  }
+
+  if (currentMonsterHealth <= 0 || currentPlayerHealth <= 0) {
+    reset();
+  }
+}
+
+function attackMonster(mode) {
+  if (mode === MODE_ATTACK) {
+    maxDamage = ATTACK_VALUE;
+    logEvent = LOG_EVENT_PLAYER_ATTACK;
+  } else if (mode === MODE_STRONG_ATTACK) {
+    maxDamage = STRONG_ATTACK_VALUE;
+    logEvent = LOG_EVENT_PLAYER_STRONG_ATTACK;
+  }
+  const damage = dealMonsterDamage(maxDamage);
+  currentMonsterHealth -= damage;
+  writeToLog(logEvent, damage, currentMonsterHealth, currentPlayerHealth);
+  endRound();
+}
+
+function attackHandler() {
+  attackMonster(MODE_ATTACK);
+}
+
+function strongAttackHandler() {
+  attackMonster(MODE_STRONG_ATTACK);
+}
+
+function healPlayerHandler() {
+  let healValue;
+  if (currentPlayerHealth >= chosenMaxLife - HEAL_VALUE) {
+    alert("You can't heal to more than your max initial health.");
+    healValue = chosenMaxLife - currentPlayerHealth;
+  } else {
+    healValue = HEAL_VALUE;
+  }
+  increasePlayerHealth(healValue);
+  currentPlayerHealth += healValue;
+  writeToLog(
+    LOG_EVENT_PLAYER_HEAL,
+    healValue,
+    currentMonsterHealth,
+    currentPlayerHealth
+  );
+  endRound();
+}
+
+function printLogHandler() {
+  for (let i = 0; i < 6; i++) {
+    console.log("------------");
+  }
+
+  let i = 0;
+  for (const logEntry of battleLog) {
```

Comment:
You could use array entries() method instead of using variable i, once you read about array methods. For now this is fine
```suggestion
  for (const [index, logEntry] of battleLog.entries()) {
```

Reply:
@coder-mezab 
@Tanvir04khan 
@Tauqeer-Ahmed-99

Reply:
okay @AlishaPrasad ma'am